From dd386504426583e68b74bcf744a8a6add9a3582a Mon Sep 17 00:00:00 2001
From: John Doe <john.doe@gmail.com>
Date: Mon, 26 Feb 2018 18:21:37 +0100
Subject: [PATCH] fix python include path

---
 src/tools/python.jam | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/src/tools/python.jam b/src/tools/python.jam
index 273b28a..b7eac5a 100644
--- a/src/tools/python.jam
+++ b/src/tools/python.jam
@@ -502,6 +502,20 @@ local rule probe ( python-cmd )
     }
 }
 
+# Use python-config command to get default paths
+#
+local rule compute-python-config-paths ( version ? : exec-prefix ? )
+{
+    local cfg-cmd = $(exec-prefix)/bin/python$(version)-config ;
+
+    local includes-cmd =
+        $(cfg-cmd)" --includes | sed -n 's/^-I\(.*\) .*/\1/p' | tr -d '\\n'" ;
+    includes ?= [ shell-cmd $(includes-cmd) ] ;
+
+    local libraries-cmd =
+        $(cfg-cmd)" --ldflags | sed -n 's/^-L\(.*\) .*/\1/p' | tr -d '\\n'" ;
+    libraries ?= [ shell-cmd $(libraries-cmd) ] ;
+}
 
 # Make sure the "libraries" and "includes" variables (in an enclosing scope)
 # have a value based on the information given.
@@ -544,6 +558,9 @@ local rule compute-default-paths ( target-os : version ? : prefix ? :
     }
     else
     {
+        # try python-config first
+        compute-python-config-paths $(version) : $(exec-prefix) ;
+
         includes ?= $(prefix)/include/python$(version) ;
 
         local lib = $(exec-prefix)/lib ;
-- 
2.7.4

